### PR TITLE
fix: (Platform) Pre-populated values not honouring the display key value

### PR DESCRIPTION
**Pull request by Lokanathan-k**
#### Please provide a link to the associated issue.
Closes :https://github.com/SAP/fundamental-ngx/issues/5147

#### Please provide a brief summary of this pull request.
fixed display key value when value pre populated

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 1]
```diff
@@ -306,7 +318,7 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
                 ? selectedItem.children[0]
                 : selectedItem
             : selectedItem;
-        this.inputText = this.displayValue(this.selected);
+        this.inputText = this.displayValue(this._selected);
```
Comment (Betrozov) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:321]:
```suggestion
        setAsSelected(items: MultiInputOption[]): void {
            this._selected = items;
            this.inputText = '';
        }
```

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 2]
```diff
@@ -65,8 +65,22 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
     listTemplateDD: ListComponent;
 
     /** Selected values from the list items. */
+
+    _selected: any[] = [];
```
Comment (Betrozov) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:69]:
```suggestion
    _selected: MultiInputOption[] = [];
```

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 3]
```diff
@@ -237,22 +249,22 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
...
     }
     /** @hidden */
     removeToken(token): void {
-        this.selected.splice(this.selected.indexOf(token), 1);
-        this.emitChangeEvent(token ? this.selected : null);
+        this._selected.splice(this._selected.indexOf(token), 1);
+        this.emitChangeEvent(token ? this._selected : null);
         this.searchInputElement.nativeElement.focus();
-        this._updateModel(this.selected);
+        this._updateModel(this._selected);
```
Comment (Betrozov) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:267]: same

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 4]
```diff
@@ -378,14 +390,14 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
...
-        this.selected = [];
+        this._selected = [];
         this.inputText = term;
         this.showList(false);
     }
 
     /** @hidden Handle dialog approval, closes popover and propagates data changes. */
     dialogApprove(): void {
-        if (this.selected && this.selectedValue.label === this.inputText) {
+        if (this._selected && this.selectedValue.label === this.inputText) {
```
Comment (Betrozov) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:400]: same

**Review by Betrozov — CHANGES_REQUESTED** [review]

**Comment by droshev** [comment]
@Lokanathan-k Can you check the title of your PR?
```
The commit contains the following structural elements, to communicate intent to the consumers of your library:

fix: a commit of the type fix patches a bug in your codebase (this correlates with PATCH in Semantic Versioning).
feat: a commit of the type feat introduces a new feature to the codebase (this correlates with MINOR in Semantic Versioning).
BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.
types other than fix: and feat: are allowed, for example @commitlint/config-conventional (based on the the Angular convention) recommends build:, chore:, ci:, docs:, style:, refactor:, perf:, test:, and others.
footers other than BREAKING CHANGE: <description> may be provided and follow a convention similar to git trailer format.

```

File: libs/platform/src/lib/components/form/multi-input/multi-input.component.ts [hunk 5]
```diff
@@ -65,8 +65,16 @@ export class PlatformMultiInputComponent extends BaseMultiInput implements OnIni
...
+    _selected: any[] = [];
+
     @Input()
-    selected: any[] = [];
+    set selected(selectedValue: any[]) {
+        this.value = selectedValue;
+    }
+    get selected(): any[] {
+        return this._selected;
+    }
```
Comment (KevinOkamoto) [inline libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:77]: Do we need `selected` as an @Input when we already have `value` as an @Input from the parent class?
https://github.com/SAP/fundamental-ngx/blob/83d8d3d2ce78a2d0639c403346cbd3c914809824/libs/platform/src/lib/components/form/multi-input/base-multi-input.ts#L134-L147

Comment (KevinOkamoto) [reply libs/platform/src/lib/components/form/multi-input/multi-input.component.ts:77]: If we need to keep `selected` then can we just leverage `value` like this?
```ts
@Input()
get selected(): any[] {
    return this.value;
}
set selected(value: any[]) {
   this.value = value;
}
```

**Review by KevinOkamoto — CHANGES_REQUESTED** [review]

**Review by KevinOkamoto — APPROVED** [review]

**Review by manu-kr — APPROVED** [review]
LGTM

**Pull request merged**